File: languages/julia/test/test_provenance.jl
using AgentBible
using JSON3
using Test

@testset "Provenance" begin
    record = ProvenanceRecord([
        CheckResult("finite_array", true, 0.0, 0.0, "n/a", nothing),
        CheckResult("unitary", true, 1e-10, 1e-12, "frobenius", nothing),
    ])

    io = IOBuffer()
    emit_provenance(record, io)
    payload = JSON3.read(String(take!(io)))

    @test payload["spec_version"] == "1.0"
    @test payload["language"] == "julia"
    @test haskey(payload, "timestamp")
    @test haskey(payload, "git_sha")
    @test length(payload["checks_passed"]) == 2
end

File: languages/rust/agentbible/tests/test_provenance.rs
use std::fs;
use std::path::PathBuf;

use agentbible::provenance::{CheckResult, ProvenanceRecord};
use serde_json::Value;

fn schema_path() -> PathBuf {
    PathBuf::from("../../../schema/provenance_v1.json")
}

#[test]
fn provenance_roundtrip_matches_schema_contract() {
    let schema: Value =
        serde_json::from_str(&fs::read_to_string(schema_path()).expect("schema")).expect("json");
    let record = ProvenanceRecord::new(vec![
        CheckResult::success("finite_array", 0.0, 0.0, "n/a"),
        CheckResult::success("unitary", 1e-10, 1e-12, "frobenius"),
    ]);
    let payload: Value = serde_json::from_str(&record.to_json_string()).expect("payload");
    let required = schema["required"].as_array().expect("required");
    for field in required {
        let name = field.as_str().expect("field");
        assert!(payload.get(name).is_some(), "missing field {name}");
    }
    assert_eq!(payload["spec_version"], "1.0");
    assert_eq!(payload["language"], "rust");
}
